src/Features/Core/Portable/FileHeaders/FileHeader.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.FileHeaders
+{
+    /// <summary>
+    /// Contains the parsed file header information for a syntax tree.
+    /// </summary>
+    internal readonly struct FileHeader
+    {
+        private readonly int _fileHeaderStart;
+        private readonly int _commentPrefixLength;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileHeader"/> struct.
+        /// </summary>
+        /// <param name="copyrightText">The copyright string, as parsed from the header.</param>
+        /// <param name="fileHeaderStart">The offset within the file at which the header started.</param>
+        /// <param name="fileHeaderEnd">The offset within the file at which the header ended.</param>
+        internal FileHeader(string copyrightText, int fileHeaderStart, int fileHeaderEnd, int commentPrefixLength)
+            : this(fileHeaderStart, isMissing: false)
+        {
+            // Currently unused
+            _ = fileHeaderEnd;
+
+            CopyrightText = copyrightText;
+            _commentPrefixLength = commentPrefixLength;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileHeader"/> struct.
+        /// </summary>
+        /// <param name="fileHeaderStart">The offset within the file at which the header started, or was expected to start.</param>
+        /// <param name="isMissing"><see langword="true"/> if the file header is missing; otherwise, <see langword="false"/>.</param>
+        private FileHeader(int fileHeaderStart, bool isMissing)
+        {
+            _fileHeaderStart = fileHeaderStart;
+            _commentPrefixLength = 0;
+
+            IsMissing = isMissing;
+            CopyrightText = "";
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the file header is missing.
+        /// </summary>
+        /// <value>
+        /// <see langword="true"/> if the file header is missing; otherwise, <see langword="false"/>.
+        /// </value>
+        internal bool IsMissing { get; }
+
+        /// <summary>
+        /// Gets the copyright text, as parsed from the header.
+        /// </summary>
+        /// <value>
+        /// The copyright text, as parsed from the header.
+        /// </value>
```

Comment:
thie 'value' portoin here just seems unhelpfully redundant.

Reply:
Keeping this one since it's a correct description of the field value.